### PR TITLE
Add TlsSocket.getProtocol() to node

### PR DESCRIPTION
**Pull request by styfle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/tls.html#tls_tlssocket_getprotocol

### CODE REVIEW

**Comment by styfle** [comment]
@andy-ms Did I cause the build failure? It looks like `node is fine but `p-event` failed.

```
p-event failing:
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/p-event/p-event-tests.ts:11:5
ERROR: 11:5  expect  TypeScript@next compile error: 
Property 'addEventListener' in type 'MyDomEmitter' is not assignable to the same property in base type 'EventTarget'.
  Type '(type: "foo", listener?: ((event: Event) => void) | undefined) => void' is not assignable to type '(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListener...'.
    Types of parameters 'listener' and 'listener' are incompatible.
      Type 'EventListenerOrEventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
        Type 'EventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
          Type 'EventListenerObject' is not assignable to type '(event: Event) => void'.
            Type 'EventListenerObject' provides no match for the signature '(event: Event): void'.
ERROR: 18:5  expect  TypeScript@next compile error: 
Property 'removeEventListener' in type 'MyDomEmitter' is not assignable to the same property in base type 'EventTarget'.
  Type '(type: "foo", listener?: ((event: Event) => void) | undefined) => void' is not assignable to type '(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | EventListenerOpt...'.
    Types of parameters 'listener' and 'listener' are incompatible.
      Type 'EventListenerOrEventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
        Type 'EventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
          Type 'EventListenerObject' is not assignable to type '(event: Event) => void'.
            Type 'EventListenerObject' provides no match for the signature '(event: Event): void'.
```

**Comment by styfle** [comment]
And another question for you:
The [node docs](https://nodejs.org/api/tls.html#tls_tlssocket_getprotocol) say that this was introduced in v5.7.0 so should I update the other files as well?

**Comment by styfle** [comment]
Did I cause the build failure? It looks like `node is fine but `p-event` failed.

```
p-event failing:
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/p-event/p-event-tests.ts:11:5
ERROR: 11:5  expect  TypeScript@next compile error: 
Property 'addEventListener' in type 'MyDomEmitter' is not assignable to the same property in base type 'EventTarget'.
  Type '(type: "foo", listener?: ((event: Event) => void) | undefined) => void' is not assignable to type '(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListener...'.
    Types of parameters 'listener' and 'listener' are incompatible.
      Type 'EventListenerOrEventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
        Type 'EventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
          Type 'EventListenerObject' is not assignable to type '(event: Event) => void'.
            Type 'EventListenerObject' provides no match for the signature '(event: Event): void'.
ERROR: 18:5  expect  TypeScript@next compile error: 
Property 'removeEventListener' in type 'MyDomEmitter' is not assignable to the same property in base type 'EventTarget'.
  Type '(type: "foo", listener?: ((event: Event) => void) | undefined) => void' is not assignable to type '(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | EventListenerOpt...'.
    Types of parameters 'listener' and 'listener' are incompatible.
      Type 'EventListenerOrEventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
        Type 'EventListenerObject' is not assignable to type '((event: Event) => void) | undefined'.
          Type 'EventListenerObject' is not assignable to type '(event: Event) => void'.
            Type 'EventListenerObject' provides no match for the signature '(event: Event): void'.
```

**Comment by Flarna** [comment]
see #23931

File: types/node/index.d.ts [hunk 1]
```diff
@@ -4895,6 +4895,14 @@ declare module "tls" {
...
         getPeerCertificate(detailed?: false): PeerCertificate;
         getPeerCertificate(detailed?: boolean): PeerCertificate | DetailedPeerCertificate;
+        /**
+         * Returns a string containing the negotiated SSL/TLS protocol version of the current connection.
+         * The value `'unknown'` will be returned for connected sockets that have not completed the handshaking process.
+         * The value `null` will be returned for server sockets or disconnected client sockets.
+         * See https://www.openssl.org/docs/man1.0.2/ssl/SSL_get_version.html for more information.
+         * @returns negotiated SSL/TLS protocol version of the current connection
+         */
+        getProtocol(): string;
```
Comment (Flarna) [inline types/node/index.d.ts]: According to docs it may also return `null` or `undefined`.

**Review by Flarna — CHANGES_REQUESTED** [review]
it would be nice to update also older versions of node typings.

**Comment by styfle** [comment]
Thank you @Flarna 🎉 

I change the type to `string | null`, applied the change on v6/v7/v8/index, and I rebased on `master`.

**Review by Flarna — APPROVED** [review]

**Pull request merged**